Replace legacy Maps embeddable with MapComponent in APM GeoMap

nickpeihl opened the pull request:
## Summary

Replaces the Maps embeddable with a new MapComponent in the APM solution.

The Maps plugin now provides an easier to use MapComponent for consumers. The legacy MapEmbeddable factory is also being removed as part of https://github.com/elastic/kibana/issues/174960 which requires making changes to consumers.


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

nickpeihl committed:
Use Maps plugin component

nickpeihl committed:
Clean up unused code and fix test

nickpeihl committed:
Merge branch 'main' into decouple-map-embeddable-apm

elasticmachine commented:
Pinging @elastic/kibana-presentation (Team:Presentation)

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

nickpeihl commented:
/ci

nickpeihl committed:
Remove unused translations

nreese commented on x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.test.tsx:
```diff
@@ -10,27 +10,15 @@ import { DataView } from '@kbn/data-views-plugin/common';
-        reload: mockReload,
-      }),
-    }));
+    const mockMapsApi = {
```
How about renaming to `mockMapsStartService`. `MapApi` is the name of the map embeddable API.

nreese commented on x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx:
```diff
@@ -36,108 +33,28 @@ function EmbeddedMapComponent({
+          setLayerList(layers);
+        } catch (e) {
+          setError(true);
+        }
```
nit, async task is not canceled and could return incorrect results if function is results return out of order. How about returning a callback, like the example below, that cancels the results from the previous call.

```
let ignore = false;

const results = await asyncStuff();
if (!ignore) {
  setLayerList(results);
}
return ()=> {
    ignore = true;
}
```

nickpeihl replied:
I was able to remove this `useEffect` in https://github.com/elastic/kibana/pull/182753/commits/c2cf3a874f62f842fb1207ca32f0434d8b629ae4 since there is no need to asynchronously load the basemap descriptor because the `MapComponent` already provides it.

nickpeihl committed:
Simplify the `EmbeddedMapComponent`

The MapComponent already provides the basemap to the embeddable so there is no need to await the basemap descriptor

nickpeihl committed:
Review feedback

nreese commented on x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx:
```diff
@@ -34,123 +29,19 @@ function EmbeddedMapComponent({
-      });
-    }
-  }, [start, end, kuery, filters, embeddable, selectedMap]);
+  const layerList = dataView?.id
```
You need a `useMemo` here otherwise, layerList will be a new array every time any props change for this Component and that will cause MapComponent to call `setLayerList` when the layerList has not changed

nickpeihl committed:
Fix container css height getting reset to 0 when changing filters

nickpeihl commented on x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx:
```diff
@@ -34,137 +29,41 @@ function EmbeddedMapComponent({
-    </>
+    <div
+      data-test-subj="serviceOverviewEmbeddedMap"
+      style={{ width: '100%', height: '500px', display: 'flex', flex: '1 1 100%', zIndex: 1 }}
```
Using `style` instead of the emotion `css` method resolves the issue that also affects the main branch where the height of this div may change to 0 when the filters change.


https://github.com/elastic/kibana/assets/1638483/286060a6-bcd0-4527-9a29-5de6ec46e650


https://github.com/elastic/kibana/assets/1638483/542ad957-31b0-4acc-a3dc-4a984900f269

nickpeihl commented on x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/map_layers/get_http_requests_map_layer_list.ts:
```diff
@@ -41,14 +41,12 @@ const label = i18n.translate('xpack.apm.serviceOverview.embeddedMap.httpRequests
     query: `${PROCESSOR_EVENT}:${ProcessorEvent.span} and ${SPAN_SUBTYPE}:${MobileSpanSubtype.Http} and ${SPAN_TYPE}:${MobileSpanType.External}`,
   };
 
-  const basemapLayerDescriptor = await maps?.createLayerDescriptors?.createBasemapLayerDescriptor();
```
It's not necessary to add the basemapLayerDescriptor as the MapComponent already takes care of that.

nickpeihl commented on x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx:
```diff
@@ -34,137 +29,41 @@ function EmbeddedMapComponent({
   filters: Filter[];
   dataView?: DataView;
```
The `dataView` passed into this component is an ad-hoc data view created from the `useAdHocApmDataView` hook. However, the ad-hoc dataview is [created without specifying a `timeFieldName`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/hooks/use_adhoc_apm_data_view.ts#L32). So Maps does not think the source is time aware and will not respond to changes to the time picker. This is a bug that currently exists in `main` as well. 

The `useAdHocApmDataView` hook is used in multiple places in this plugin so I did not feel comfortable making any changes there.

nickpeihl committed:
Review feedback

nreese reviewed (approved):
LGTM
code review only

nickpeihl committed:
Merge branch 'main' into decouple-map-embeddable-apm

cauemarcondes reviewed (approved):
LGTM
